znft/ethereum_wallet.go

```diff
@@ -0,0 +1,93 @@
+package znft
+
+import (
+	"context"
+	"math/big"
+	"path"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/pkg/errors"
+)
+
+func (conf *Configuration) CreateEthClient() (*ethclient.Client, error) {
+	client, err := ethclient.Dial(conf.EthereumNodeURL)
+	if err != nil {
+		Logger.Error(err)
+	}
+	return client, err
+}
+
+func (conf *Configuration) createSignedTransactionFromKeyStore() *bind.TransactOpts {
+	var (
+		signerAddress = common.HexToAddress(conf.WalletAddress)
+		password      = conf.VaultPassword
+		value         = conf.Value
+	)
+
+	client, err := conf.CreateEthClient()
+	if err != nil {
+		Logger.Fatal(errors.Wrap(err, "failed to create ethereum client"))
+	}
+
+	keyDir := path.Join(conf.Homedir, WalletDir)
+	ks := keystore.NewKeyStore(keyDir, keystore.StandardScryptN, keystore.StandardScryptP)
+	signer := accounts.Account{
+		Address: signerAddress,
+	}
+	signerAcc, err := ks.Find(signer)
+	if err != nil {
+		Logger.Fatal(errors.Wrapf(err, "signer: %s", signerAddress.Hex()))
+	}
+
+	chainID, err := client.ChainID(context.Background())
```

Comment:
We'd better not use context.Background(), instead, let's use context passed by the user, i.e add `ctx context.Context` param to the `createSignedTransactionFromKeyStore` function, so as other functions below require context.

Reply:
Fixed in 6530477dc45ee27386c2d579e3feca85c1084d4c